Shared (App)/SettingsTab/SettingsView.swift

```diff
@@ -129,16 +137,16 @@ extension SettingsView {
         @Published var networkIndex: Int = 0 {
             didSet {
                 print("network set to \(networkIndex)")
-//                guard oldValue != networkIndex else { return }
-//                Task {
-//                    let network: Network
-//                    if networkIndex == 0 {
-//                        network = .ethereum
-//                    } else {
-//                        network = .ropsten
-//                    }
-//                    await change(network: network)
-//                }
+                guard oldValue != networkIndex else { return }
+                Task {
+                    let network: Network
+                    if networkIndex == 0 {
+                        network = .ethereum
+                    } else {
+                        network = .ropsten
+                    }
+                    await change(network: network)
```

Comment:
Could it be an issue when you set the networkIndex and try to access bundles or addresses before they have been loaded (since change(network:) is an async call)? If we made networkIndex private(set) and only allowed changing networks through `change(network:) async` then the caller can be sure that bundles/addresses have been fully loaded after awaiting the call.

Reply:
Yes, that is what I originally did but that required a lot of glue code to link to SwiftUI (that may be because I don't not too experienced in SwiftUI yet, I might have missed something). This way it does work, there's just a slight delay related to file access: since both the extension and the app can write to the same files, there's a queue.